### PR TITLE
added logic to filter by platform name (genotyping and sequencing)

### PR DESCRIPTION
**Pull request by samouzegar10**
decided to go with filters instead of separating by tab

### CODE REVIEW

**Comment by samouzegar10** [comment]
also, some code refactor to make the code look prettier :)

**Comment by nafiz1001** [comment]
Having whitespace changes can slow down PR review. I believe making the code prettier should be a dedicated PR.

**Comment by UlysseFG** [comment]
I'm not completely sold to the dropdown approach i would prefer the radio group seen in samples but a couple of things of more importance :
- The filter need to be shown at all time (no collapsible).
- The default needs to be Illumina - Sequencing, not All.
- There should be another filter for the progress of the dataset release... Not validated (default), Validated, Released, All.
- The Clear Filter option should be to the right of the table like for other tables.

File: frontend/src/components/experimentRuns/ExperimentRunsDetachedFilters.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { FilterDescription } from "../../models/paged_items";
+import { FILTER_TYPE } from '../../constants'
+
+export const EXPERIMENT_RUNS_PLATFORM_NAME_FILTER: FilterDescription = {
+    type: FILTER_TYPE.SELECT,
+    key: 'run_type__platform__name',
+    label: 'Platfor Name',
```
Comment (UlysseFG) [inline frontend/src/components/experimentRuns/ExperimentRunsDetachedFilters.ts:7]: "Platform"

File: frontend/src/components/experimentRuns/ExperimentRunsTabs.tsx [hunk 2]
```diff
@@ -28,21 +28,22 @@ const tabStyle = {
...
+function ExperimentRunsTabs() {
+	const [activeKey, setActiveKey] = useHashURL(ExperimentRunsTabsName.FREEZEMAN)
 	const token = useAppSelector(selectAuthTokenAccess)
 	const experimentRunsTableState = useAppSelector(selectExperimentRunsTable)
 	const actions = useAppSelector(selectExperimentRunsTemplateActions)
 
 
 	function getPageHeaderExtra() {
-		if (activeKey === FREEZEMAN_TAB_KEY) {
+		if (activeKey === ExperimentRunsTabsName.FREEZEMAN) {
```
Comment (nafiz1001) [inline frontend/src/components/experimentRuns/ExperimentRunsTabs.tsx:46]: I think this function should be in a `useCallback`.

File: frontend/src/utils/api.ts
Comment (nafiz1001) [inline frontend/src/utils/api.ts:1]: I love prettier codes, but this is too many white-space changes to look at. Again, it should be a separate PR.

Comment (nafiz1001) [reply frontend/src/utils/api.ts:1]: TBF. GitHub let's you hide white-space changes. So maybe you can keep this change.

**Review by nafiz1001 — APPROVED** [review]
Besides what Ulysse said, I approve.

**Comment by samouzegar10** [comment]
> I'm not completely sold to the dropdown approach i would prefer the radio group seen in samples but a couple of things of more importance :
> 
> * The filter need to be shown at all time (no collapsible).
> * The default needs to be Illumina - Sequencing, not All.
> * There should be another filter for the progress of the dataset release... Not validated (default), Validated, Released, All.
> * The Clear Filter option should be to the right of the table like for other tables.

these have all been resolved

File: frontend/src/components/experimentRuns/ExperimentRunsTabs.tsx (hunk 2 above)
Comment (samouzegar10) [reply frontend/src/components/experimentRuns/ExperimentRunsTabs.tsx:46]: this was initially part of the changes but after bashing my head on this task for several days, i decided to simply leave the enum instead of the initial freezeman key.

File: backend/fms_core/viewsets/_constants.py [hunk 3]
```diff
@@ -153,20 +153,7 @@
...
-    "start_date": DATE_FILTERS,
-    "run_type": FK_FILTERS,
-    "instrument": FK_FILTERS,
-    "container": FK_FILTERS,
-    "run_processing_launch_time": DATE_FILTERS,
-
-    **_prefix_keys("run_type__", _run_type_filterset_fields),
-    **_prefix_keys("instrument__", _instrument_filterset_fields),
-    **_prefix_keys("container__", _container_filterset_fields),
+    **_prefix_keys("platform__", _platform_filterset_fields),
```
Comment (UlysseFG) [inline backend/fms_core/viewsets/_constants.py:156]: Run type name might actually be a better identifier for the grouping ... You could display Illumina, Axiom, DNBSEQ, Infinium straight  out of the DB. You can leave the filterset field for the platform but it might actually be one additional indirection we might not need.

File: backend/fms_core/viewsets/_constants.py [hunk 4]
```diff
@@ -242,6 +229,24 @@
...
+    "readsets__validation_status": CATEGORICAL_FILTERS_LOOSE,
+    "readsets__release_status": CATEGORICAL_FILTERS_LOOSE,
+
+}
+
+_experiment_run_filterset_fields: FiltersetFields = {
+    "id": PK_FILTERS,
+    "name": CATEGORICAL_FILTERS_LOOSE,
+    "start_date": DATE_FILTERS,
+    "run_type": FK_FILTERS,
```
Comment (UlysseFG) [inline backend/fms_core/viewsets/_constants.py:241]: If you break down the object in the filterset remove the FK filterset ... and make sure it is no longer referenced in the frontend...

File: backend/fms_core/viewsets/_constants.py [hunk 5]
```diff
@@ -242,6 +229,24 @@
...
+    "instrument": FK_FILTERS,
+    "container": FK_FILTERS,
+    "run_processing_launch_time": DATE_FILTERS,
+
+    **_prefix_keys("run_type__", _run_type_filterset_fields),
+    **_prefix_keys("instrument__", _instrument_filterset_fields),
+    **_prefix_keys("container__", _container_filterset_fields),
+    **_prefix_keys("datasets__", _dataset_filterset_fields),
```
Comment (UlysseFG) [inline backend/fms_core/viewsets/_constants.py:249]: All these filterset might be a bit overkill for the need we have ... We basically want to know if the experiment_run is processed (at least one dataset validation at available), validated (No dataset validation available, all validation status either passed or failed), released (all released status at released or blocked). This much business logic does not sit well in a frontend ... we can make a backend filter for experiment_run progression state. The additional filters for release or blocked, passed or failed would be better elsewhere ... they would be part of another PR.

**Comment by UlysseFG** [comment]
All the filters are clumped together to the right ... they don't seem to be at the right place ... Also they do not feel right ... At the experiment run level it does not really make sense to start filtering by detailed status. User at this point want to know where are the experiment_runs i have not validated, i have not released ... once they get validated released the exploration of detailed status should be done in the dataset section at another time. So i suggest we replace all the filtersets added by a simple custom filter on experiment run that will filter on the general progress of the processing.

File: frontend/src/components/experimentRuns/ExperimentRunsDetachedFilters.ts [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+import { FILTER_TYPE } from '../../constants'
+
+export const EXPERIMENT_RUNS_PLATFORM_NAME_FILTER: FilterDescription = {
+    type: FILTER_TYPE.SELECT,
+    key: 'run_type__platform__name',
+    label: 'Platform Name',
+    mode: 'multiple',
+    placeholder: 'All',
+    options: [
+        { label: 'Axiom - Genotyping', value: 'AXIOM_ARRAY' },
```
Comment (samouzegar10) [inline frontend/src/components/experimentRuns/ExperimentRunsDetachedFilters.ts:11]: remove genotyping and sequencing word in filter label

**Comment by UlysseFG** [comment]
Rebase your code so there is no conflict please.

**Comment by UlysseFG** [comment]
The visual is great. The Progress filter is not working as intended yet ... It should default to processed for one. Also once an experiment pass to the next stage it should no longer appear in the other stages. At the moment, filtering for any stage returns the same experiments.

File: backend/fms_core/filters.py [hunk 7]
```diff
@@ -193,6 +197,21 @@ class Meta:
...
         fields = _dataset_filterset_fields
 
+class ExperimentRunFilter(GenericFilter):
+    experiment_run_progress_stage = django_filters.CharFilter(method="experiment_run_progress_stage_filter")
+
+    def experiment_run_progress_stage_filter(self, queryset, name, value):
+        if value == "processed":
+            return queryset.filter(datasets__readsets__validation_status__in=[ValidationStatus.PASSED.value,ValidationStatus.FAILED.value])
+        if value == "validated":
+            return queryset.filter(datasets__readsets__release_status=ReleaseStatus.AVAILABLE.value,datasets__readsets__validation_status__in=[ValidationStatus.PASSED.value,ValidationStatus.FAILED.value])
```
Comment (UlysseFG) [inline backend/fms_core/filters.py:207]: you can use Q queries with queryset.exists(...) statements for the AVAILABLE test. Those are pretty fasts and are easy to read...

File: frontend/src/components/experimentRuns/ExperimentRunsDetachedFilters.ts [hunk 8]
```diff
@@ -0,0 +1,33 @@
+import { FilterDescription } from "../../models/paged_items";
+import { FILTER_TYPE } from '../../constants'
+import { capitalize } from "../../utils/capitalize";
```
Comment (samouzegar10) [inline frontend/src/components/experimentRuns/ExperimentRunsDetachedFilters.ts:3]: i like how the code looks, the only error i got was this capitalize function/file not appearing. the fix is easy as well.

i only removed the function and added flag.toUpperCase() vs capitalize(flag)

Comment (UlysseFG) [reply frontend/src/components/experimentRuns/ExperimentRunsDetachedFilters.ts:3]: You can remove the capitalize file then if there is some built-in function that does that ...
